### PR TITLE
refactor: integrations & apiservice

### PR DESCRIPTION
**Pull request by FabianKramm**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

Changes:
* Refactored import / export syncer
* Refactored metrics-server integration
* Renamed `observability.proxy.metrics` -> `integrations.metricsServer`

### CODE REVIEW

File: pkg/apiservice/generic.go [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+	s := serializer.NewCodecFactory(scheme.Scheme)
+	server := &http.Server{
+		Addr: "localhost:" + strconv.Itoa(proxyPort),
+		Handler: http.HandlerFunc(func(writer http.ResponseWriter, request *http.Request) {
+			// we only allow traffic to discovery paths
+			if !isAPIServiceProxyPathAllowed(request.Method, request.URL.Path) {
+				klog.FromContext(ctx).Info("Denied access to api service proxy at path", "path", request.URL.Path, "method", request.Method)
+				responsewriters.ErrorNegotiated(
+					kerrors.NewForbidden(metav1.SchemeGroupVersion.WithResource("proxy").GroupResource(), "proxy", fmt.Errorf("other paths than discovery paths are not allowed")),
```
Comment (zerbitx) [inline pkg/apiservice/generic.go:158]: nit:
```suggestion
					kerrors.NewForbidden(metav1.SchemeGroupVersion.WithResource("proxy").GroupResource(), "proxy", fmt.Errorf("paths other than discovery paths are not allowed")),
```

File: pkg/util/unstructuredhelper/write.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+	lastKey := pathSplitted[len(pathSplitted)-1]
+	if !targetMap.Has(lastKey) {
+		return
+	}
+
+	// to map
+	toMap := to
+	for i := 0; i < len(pathSplitted)-1; i++ {
+		toMap = toMap.Map(pathSplitted[i])
+	}
```
Comment (zerbitx) [inline pkg/util/unstructuredhelper/write.go:89]: nit: traverses pathSplitted only once.

```suggestion
	targetMap := from
	// to map
	toMap := to
	for i := range pathSplitted {
		targetMap = targetMap.Map(pathSplitted[i])
		toMap = toMap.Map(pathSplitted[i])
	}

	// check if map has latest path
	lastKey := pathSplitted[len(pathSplitted)-1]
	if !targetMap.Has(lastKey) {
		return
	}
```

**Review by zerbitx — APPROVED** [review]

File: pkg/util/unstructuredhelper/write.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+
+	// check if map has latest path
+	lastKey := pathSplitted[len(pathSplitted)-1]
+
+	// to map
+	toMap := to
+	for i := 0; i < len(pathSplitted)-1; i++ {
+		toMap = toMap.Map(pathSplitted[i])
+	}
```
Comment (zerbitx) [inline pkg/util/unstructuredhelper/write.go:55]: nit: avoids second traversal

```suggestion
	targetMap := from
	// to map
	toMap := to
	for i := 0; i < len(pathSplitted)-1; i++ {
		targetMap = targetMap.Map(pathSplitted[i])
		toMap = toMap.Map(pathSplitted[i])
	}

	// check if map has latest path
	lastKey := pathSplitted[len(pathSplitted)-1]
```

File: chart/values.yaml [hunk 4]
```diff
@@ -116,6 +116,17 @@ sync:
         all: false
         labels: {}
 
+# Integrations holds config for vCluster integrations with other operators or tools
```
Comment (johannesfrey) [inline chart/values.yaml:119]:
```suggestion
# Integrations holds config for vCluster integrations with other operators or tools running on the host cluster
```
Not sure if this suggestion narrows down the use case of `integrations` too much, but I could see that users might be confused how this is different to `external` then.

Comment (FabianKramm) [reply chart/values.yaml]: Makes sense, external is basically configurations for tooling outside of vCluster and integrations are part of vCluster itself

**Pull request merged**